client/my-sites/email/titan-mail-quantity-selection/index.jsx

```diff
@@ -0,0 +1,217 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import page from 'page';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Card } from '@automattic/components';
+import DomainManagementHeader from 'calypso/my-sites/domains/domain-management/components/header';
+import Main from 'calypso/components/main';
+import SectionHeader from 'calypso/components/section-header';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import { emailManagement } from 'calypso/my-sites/email/paths';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { recordTracksEvent as recordTracksEventAction } from 'calypso/state/analytics/actions';
+import { addItems } from 'calypso/lib/cart/actions';
+import { titanMailMonthly } from 'calypso/lib/cart-values/cart-items';
+import {
+	getDomainsBySiteId,
+	hasLoadedSiteDomains,
+	isRequestingSiteDomains,
+} from 'calypso/state/sites/domains/selectors';
+import { getDomainsWithForwards } from 'calypso/state/selectors/get-email-forwards';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import Notice from 'calypso/components/notice';
+import AddEmailAddressesCardPlaceholder from 'calypso/my-sites/email/gsuite-add-users/add-users-placeholder';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { hasTitanMailWithUs, getMaxTitanMailboxCount } from 'calypso/lib/titan';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class TitanMailQuantitySelection extends React.Component {
+	state = {
+		quantity: 1,
+	};
+
+	recordClickEvent = ( eventName ) => {
+		const { recordTracksEvent, selectedDomainName } = this.props;
+		recordTracksEvent( eventName, {
+			domain_name: selectedDomainName,
+			quantity: this.state.quantity,
+		} );
+	};
+
+	goToEmail = () => {
+		page(
+			emailManagement(
+				this.props.selectedSite.slug,
+				this.props.selectedDomain?.name,
+				this.props.currentRoute
+			)
+		);
+	};
+
+	handleCancel = () => {
+		this.recordClickEvent(
+			'calypso_email_management_titan_quantity_selection_cancel_button_click'
+		);
+		this.goToEmail();
+	};
+
+	getCartItem = () => {
+		const { selectedDomain } = this.props;
+		let quantity = this.state.quantity;
+
+		if ( hasTitanMailWithUs( selectedDomain ) ) {
+			quantity += getMaxTitanMailboxCount( selectedDomain );
+		}
+
+		return titanMailMonthly( { domain: selectedDomain.name, quantity } );
+	};
+
+	handleContinue = () => {
+		const { selectedSite } = this.props;
+
+		this.recordClickEvent(
+			'calypso_email_management_titan_quantity_selection_continue_button_click'
+		);
+
+		addItems( [ this.getCartItem() ] );
```

Comment:
As we're trying to delete the `CartStore` (see https://github.com/Automattic/wp-calypso/issues/24019) and replace it with the `@automattic/shopping-cart` functionality, could we use `addProductsToCart()` instead of the deprecated `addItems`? You'll need to wrap this component in the `withShoppingCart` HOC first and make sure that one of this component's parents includes `CalypsoShoppingCartProvider`.

```suggestion
		this.props.shoppingCartManager.addProductsToCart( [ fillInSingleCartItemAttributes( this.getCartItem() ) ] );
```

See https://github.com/Automattic/wp-calypso/pull/48202 for an example of this pattern.

Reply:
Will do. Thank you for the heads up!